src/components/Namespaces/NamespaceSidebar.vue

```diff
@@ -128,22 +178,99 @@ export default {
 
     adminRoutes () {
       return [
-        { name: 'admin.modules', title: this.$t('module') },
-        { name: 'admin.pages', title: this.$t('page') },
-        { name: 'admin.charts', title: this.$t('chart') },
+        {
+          page: {
+            pageID: 'modules',
+            selfID: NoID,
+            name: 'admin.modules',
+            title: this.$t('module'),
+            visible: true,
+          },
+          children: [],
+        },
+        ...this.modules.map(moduleWrap),
+        {
+          page: {
+            pageID: 'pages',
+            selfID: NoID,
+            name: 'admin.pages',
+            title: this.$t('page'),
+            visible: true,
+          },
+          children: [],
+        },
+        ...this.pages.map(adminPageWrap),
+        {
+          page: {
+            pageID: 'charts',
+            selfID: NoID,
+            name: 'admin.charts',
+            title: this.$t('chart'),
+            visible: true,
+          },
+          children: [],
+        },
+        ...this.charts.map(chartWrap),
       ]
     },
 
-    navItems () {
-      let wrap = (page) => ({ page, children: [], params: { pageID: page.pageID } })
+    publicRoutes () {
+      return this.pages.filter(({ moduleID, visible }) => visible && moduleID === NoID)
+    },
 
-      if (this.isAdminPage) {
-        wrap = (page) => ({ page, children: [], params: {} })
+    filteredPages () {
+      if (this.namespace) {
+        // If on admin page, show admin pages. Otherwise show public pages
+        const pages = this.isAdminPage ? this.adminRoutes : this.publicRoutes.map(publicPageWrap)
+
+        if (!this.query) {
+          return pages
+        }
+
+        return pages.filter(({ page }) => `${page.title}`.toLowerCase().trim().includes(this.query))
       }
 
-      const current = this.filteredPages.map(wrap)
+      return []
+    },
+
+    truncatedNamespaces () {
+      return this.namespaces.map(ns => ns.name.length > 21 ? { ...ns, name: ns.name.substring(0, 21) + '...' } : ns)
```

Comment:
Why are we adding ellipsis with JS and not with CSS?

Reply:
plus; if doing this in the JS is the only way -- how about using the `namespaces` and just passing the name through a JS function to shorten it while looping over the array in the templatE?

Reply:
Was introduced previously so should be fixed in another commit not this PR
Same for the typo